spec/datadog/appsec/configuration/settings_spec.rb

```diff
@@ -0,0 +1,192 @@
+      it { is_expected.to eq(true) }
+    end
+
+    describe '#enabled=' do
+      subject(:enabled_) { settings.merge(dsl.tap { |c| c.enabled = false }) }
+      it { expect { enabled_ }.to change { settings.enabled }.from(true).to(false) }
+    end
+
+    describe '#ruleset' do
+      subject(:ruleset) { settings.ruleset }
+      it { is_expected.to eq(:recommended) }
+    end
+
+    describe '#ruleset=' do
+      subject(:ruleset_) { settings.merge(dsl.tap { |c| c.ruleset = :risky }) }
+      it { expect { ruleset_ }.to change { settings.ruleset }.from(:recommended).to(:risky) }
+    end
+
+    describe '#waf_timeout' do
+      subject(:waf_timeout) { settings.waf_timeout }
+      it { is_expected.to eq(5000) }
+    end
+
+    describe '#waf_timeout=' do
+      subject(:waf_timeout_) { settings.merge(dsl.tap { |c| c.waf_timeout = 3 }) }
+      it { expect { waf_timeout_ }.to change { settings.waf_timeout }.from(5000).to(3) }
+    end
+
+    describe '#waf_debug' do
+      subject(:waf_debug) { settings.waf_debug }
+      it { is_expected.to eq(false) }
+    end
+
+    describe '#waf_debug=' do
+      subject(:waf_debug_) { settings.merge(dsl.tap { |c| c.waf_debug = true }) }
+      it { expect { waf_debug_ }.to change { settings.waf_debug }.from(false).to(true) }
+    end
+
+    describe '#trace_rate_limit' do
+      subject(:trace_rate_limit) { settings.trace_rate_limit }
+      it { is_expected.to eq(100) }
+    end
+
+    describe '#trace_rate_limit=' do
+      subject(:trace_rate_limit_) { settings.merge(dsl.tap { |c| c.trace_rate_limit = 2 }) }
+      it { expect { trace_rate_limit_ }.to change { settings.trace_rate_limit }.from(100).to(2) }
+    end
+
+    describe '#[]' do
+      describe 'when the integration exists' do
+        subject(:get) { settings[integration_name] }
+
+        let(:integration_options) { { foo: :bar } }
+
+        before { settings.merge(dsl.tap { |c| c.instrument(integration_name, integration_options) }) }
+
+        it 'retrieves the described configuration' do
+          is_expected.to eq(integration_options)
+        end
+      end
+
+      context 'when the integration doesn\'t exist' do
+        it do
+          expect { settings[:foobar] }.to raise_error(ArgumentError, /foobar/)
+        end
+      end
+    end
+
+    context 'with env vars' do
+      before do
+        stub_const('ENV', {})
+        allow(ENV).to receive(:[]).and_call_original
+      end
+
+      describe 'DD_APPSEC_ENABLED' do
+        before do
+          allow(ENV).to receive(:[]).with('DD_APPSEC_ENABLED').and_return('1')
+        end
+
+        describe '#enabled' do
+          subject(:enabled) { settings.enabled }
+          it { is_expected.to eq(true) }
+        end
+
+        describe '#enabled=' do
+          subject(:enabled_) { settings.merge(dsl.tap { |c| c.enabled = false }) }
+          it { expect { enabled_ }.to change { settings.enabled }.from(true).to(false) }
+        end
```

Comment:
The tested behaviour here is that the env var should:

a) have less priority than the DSL setters (final value)
b) still be picked up before the DSL kicks in (change, notably from a non-default to a non-default)

This was actually useful to test because I triggered a small bug with it (that affected only specs, not real life apps) when this example was in `spec/datadog/appsec/extensions_spec.rb`.